Automattic/jetpack
# Search: Use library function for number formatting.

#### Changes proposed in this Pull Request:


Uses the library function for number formatting. This takes into account the user's locale if available and keeps things consistent across the Search package.

#### Other information:


#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






1. Set up a test site with Jetpack installed.
2. Add a Search plan.
3. Change the site language to French.
4. Install the updated language files from `/wp-admin/update-core.php`. Only visible after changing the site language.
5. Visit the dashboard at `/wp-admin/admin.php?page=jetpack-search&new_pricing_202208=1`.
6. Confirm the numbers are formatted to the site locale (in this case, French) instead of to the browser locale.
